src/Microsoft.AspNetCore.Mvc.Core/Internal/DefaultApplicationModelProvider.cs

```diff
@@ -395,10 +395,7 @@ protected virtual PropertyModel CreatePropertyModel(PropertyInfo propertyInfo)
                 // public void Foo() { }
                 //
                 // Is one action.
-                if (!(routeProvider is IActionHttpMethodProvider))
-                {
-                    createActionForSilentRouteProviders = false;
-                }
+                createActionForSilentRouteProviders &= routeProvider is IActionHttpMethodProvider;
```

Comment:
Not personally a fan of the readability or debuggability of this (it was easy to set a breakpoint in the `if()` before).